src/wp-includes/class-wp-block.php

```diff
@@ -333,6 +333,74 @@ private function replace_html( string $block_content, string $attribute_name, $s
 		switch ( $block_type->attributes[ $attribute_name ]['source'] ) {
 			case 'html':
 			case 'rich-text':
+				if ( 'core/image' === $this->name && 'caption' === $attribute_name ) {
+					// Create private anonymous class until the HTML API provides `set_inner_html` method.
+					$bindings_processor = new class( $block_content, WP_HTML_Processor::CONSTRUCTOR_UNLOCK_CODE ) extends WP_HTML_Processor {
+						/**
+						 * Replace the inner content of a figcaption element with the passed content.
+						 *
+						 * THIS IS A TEMPORARY SOLUTION IN CORE NOT TO BE EMULATED.
+						 * IT IS A TEMPORARY SOLUTION THAT JUST WORKS FOR THIS SPECIFIC
+						 * USE CASE UNTIL THE HTML PROCESSOR PROVIDES ITS OWN METHOD.
```

Comment:
"Temporary" is a funny word in codebases, as it tends not to be, and it doesn't provide information for future decisions. I like the part indicating that this should be replaced in the future by the HTML Processor's own method.

Maybe we can emphasize this mor along those lines.

> DO NOT COPY THIS METHOD
> THE HTML PROCESSOR WILL HAVE A PROPER METHOD
> USE IT INSTEAD